packages/@aws-cdk/aws-codestarnotifications/test/notification-rule.test.ts

```diff
@@ -0,0 +1,196 @@
+      Resource: 'arn:aws:codebuild::1234567890:project/MyCodebuildProject',
+    });
+  });
+
+  test('created new notification rule with status DISABLED', () => {
+    const project = new FakeCodeBuild();
+
+    new notifications.NotificationRule(stack, 'MyNotificationRule', {
+      source: project,
+      enabled: false,
+    });
+
+    expect(stack).toHaveResourceLike('AWS::CodeStarNotifications::NotificationRule', {
+      Name: 'MyNotificationRule',
+      Resource: 'arn:aws:codebuild::1234567890:project/MyCodebuildProject',
+      Status: 'DISABLED',
+    });
+  });
+
+  test('created new notification rule with status ENABLED', () => {
+    const project = new FakeCodeBuild();
+
+    new notifications.NotificationRule(stack, 'MyNotificationRule', {
+      source: project,
+      enabled: true,
+    });
+
+    expect(stack).toHaveResourceLike('AWS::CodeStarNotifications::NotificationRule', {
+      Name: 'MyNotificationRule',
+      Resource: 'arn:aws:codebuild::1234567890:project/MyCodebuildProject',
+      Status: 'ENABLED',
+    });
+  });
+
+  test('notification added targets', () => {
+    const project = new FakeCodeBuild();
+    const topic = new FakeSnsTopicTarget();
+    const slack = new FakeSlackTarget();
+
+    const rule = new notifications.NotificationRule(stack, 'MyNotificationRule', {
+      source: project,
+    });
+
+    rule.addTarget(slack);
+
+    expect(rule.addTarget(topic)).toEqual(true);
+
+    expect(stack).toHaveResourceLike('AWS::CodeStarNotifications::NotificationRule', {
+      Resource: 'arn:aws:codebuild::1234567890:project/MyCodebuildProject',
+      Targets: [
+        {
+          TargetAddress: 'arn:aws:chatbot::1234567890:chat-configuration/slack-channel/MySlackChannel',
+          TargetType: 'AWSChatbotSlack',
+        },
+        {
+          TargetAddress: 'arn:aws:sns::1234567890:MyTopic',
+          TargetType: 'SNS',
+        },
+      ],
+    });
+  });
+
+  test('will not effect and return false when added targets if notification from imported', () => {
+    const imported = notifications.NotificationRule.fromNotificationRuleArn(stack, 'MyNotificationRule', 'arn:aws:codestar-notifications::1234567890:notificationrule/1234567890abcdef');
+    const slack = new FakeSlackTarget();
+    expect(imported.addTarget(slack)).toEqual(false);
+  });
+
+  test('will not add if notification added duplicating event', () => {
+    const pipeline = new FakeCodePipeline();
+    new notifications.NotificationRule(stack, 'MyNotificationRule', {
+      source: pipeline,
+      events: [
+        'codepipeline-pipeline-pipeline-execution-succeeded',
+        'codepipeline-pipeline-pipeline-execution-failed',
+        'codepipeline-pipeline-pipeline-execution-succeeded',
+        'codepipeline-pipeline-pipeline-execution-canceled',
+      ],
+    });
+
+    expect(stack).toHaveResourceLike('AWS::CodeStarNotifications::NotificationRule', {
+      Resource: 'arn:aws:codepipeline::1234567890:MyCodepipelineProject',
```

Comment:
Same note about the random hard-coded ARNs being present in test assertions here.

Reply:
Already adjusted.